### PR TITLE
Feat - Implement Cereal endpoints

### PR DESCRIPTION
**Pull request by mirycantero**
#### What does this PR do?

Implement endpoints to consult a CloudSQL database.
The database only contains one table `cereal`.
The implemented endpoints provide the following functionality:
- List all active cereals
- Get a cereal
- Create a cereal
- Update a cereal
- Delete a cereal

#### Where should the reviewer start?

The reviewer should start looking at the `app/endpoints/cereals.py` file

#### How should this be manually tested?

1. Run the application (Please refer to the `Quick start` section in the README file)
2. Ping the following endpoints:
`GET http://localhost:8080/cereals`
`GET http://localhost:8080/cereals/{name}`
`POST http://localhost:8080/cereals`
POST Request body:
```
{
    "calories": float,
    "carbo": float,
    "fat": float,
    "fiber": float,
    "name": string,
    "potass": float,
    "protein": float,
    "sodium": float,
    "sugars": float,
    "vitamins": float
}
```
`PUT http://localhost:8080/cereals/{name}`
PUT Request body:
```
{
    "calories": float,
    "carbo": float,
    "fat": float,
    "fiber": float,
    "potass": float,
    "protein": float,
    "sodium": float,
    "sugars": float,
    "vitamins": float
}
```
`DELETE http://localhost:8080/cereals/{name}`

#### Any background context you want to provide?

The Cereal dataset was downloaded from https://www.kaggle.com/crawford/80-cereals/data

### CODE REVIEW

File: app/endpoints/cereals.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+CORS(cereals)
+
+
+@cereals.route('/cereals', methods=['GET'])
+def get_cereals():
+    cereals = Cereal.get_cereals()
+    return custom_response(cereals, 200)
+
+
+@cereals.route('/cereals/<string:name>', methods=['GET'])
```
Comment (timgivois) [inline app/endpoints/cereals.py]: what happens if I try to search for a cereal that doesn't exist? in http rest apis I should receive a 404

File: app/endpoints/cereals.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    cereal = Cereal.get_cereal(name)
+    return custom_response(cereal, 200)
+
+
+@cereals.route('/cereals', methods=['POST'])
+def post_cereal():
+    cereal_payload = request.json
+
+    cereal = Cereal.post_cereal(cereal_payload)
+    return custom_response(cereal, 200)
```
Comment (timgivois) [inline app/endpoints/cereals.py]: the answer should be 201

Comment (timgivois) [reply app/endpoints/cereals.py]: Plus, what happens if I don't pass the required payload

File: app/endpoints/cereals.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+
+@cereals.route('/cereals/<string:name>', methods=['PUT'])
+def put_cereal(name):
+    cereal_payload = request.json
+
+    cereal = Cereal.put_cereal(name, cereal_payload)
+    return custom_response(cereal, 200)
```
Comment (timgivois) [inline app/endpoints/cereals.py]: 204 is the code for successful PUT, and you should sent another error code

File: app/endpoints/cereals.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+
+@cereals.route('/cereals/<string:name>', methods=['DELETE'])
+def delete_cereal(name):
+    cereal = Cereal.delete_cereal(name)
+    return custom_response(cereal, 200)
```
Comment (timgivois) [inline app/endpoints/cereals.py]: if successful, delete should be 204

File: app/endpoints/cereals.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    return custom_response(cereals, 200)
+
+
+@cereals.route('/cereals/<string:name>', methods=['GET'])
+def get_cereal(name):
+    cereal = Cereal.get_cereal(name)
+    return custom_response(cereal, 200)
+
+
+@cereals.route('/cereals', methods=['POST'])
```
Comment (timgivois) [inline app/endpoints/cereals.py]: this should be `cereal`

**Comment by oscarSantoyo** [comment]
I believe that the recommendations are not a blocker to merge this PR...
Let's merge it.

**Pull request merged**